src/bluesky_stomp/messaging.py

```diff
@@ -384,10 +401,18 @@ def _on_message(self, frame: Frame) -> None:
             Mapping[str, Any],
             frame.headers,  # type: ignore
         )
+
+        trace_context = retrieve_context_from_stomp_headers(frame)
+
+        tracer = get_tracer("_on_message")
+
         if (sub_id := headers.get("subscription")) is not None:
             if (sub := self._subscriptions.get(sub_id)) is not None:
                 try:
-                    sub.callback(frame)
+                    with tracer.start_as_current_span(
+                        sub.callback.__name__, trace_context
```

Comment:
I would use _on_message rather than sub.callback.__name__ here, since that's where we are, and then add spans around to call back methods that can be called. That way, we gat a better picture of whethere the correct mesages asre being routed to the correct callback. Also we should add the frame as a span attribute.